cmd/backup_resource.go

```diff
@@ -0,0 +1,228 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus"
+	"github.com/odpf/optimus/config"
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/salt/log"
+	"github.com/pkg/errors"
+	cli "github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+func backupResourceSubCommand(l log.Logger, datastoreRepo models.DatastoreRepo, conf config.Provider) *cli.Command {
+	var (
+		backupCmd = &cli.Command{
+			Use:   "resource",
```

Comment:
@kushsharma IMO, `backup` command should take resource as arg instead of sub command. If user does not pass  resource then we can promote them. 
`backup <id>` WDYT?

Reply:
Another way could be to call it `run` instead of `resource`.

Reply:
term `resource` is currently used for BQ table, Postgres table, GCS bucket, etc whereas `run` will be used for job instances. I think I am confused about what you trying to say, can you elaborate a little more?

Reply:
`backup resource` sounds like two nous.

if we are treating backup as noun. then command can be `backup create`  or `backup run`

Reply:
Makes perfect sense, changed it `optimus backup create --resource "resource_name"`